bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -42,18 +49,29 @@
  * @author Sebastian Prehn - Initial contribution (v1.x hub communication)
  *
  */
+@NonNullByDefault
 public class NeoHubHandler extends BaseBridgeHandler {
 
     private final Logger logger = LoggerFactory.getLogger(NeoHubHandler.class);
 
-    private NeoHubConfiguration config;
-    private NeoHubSocket socket;
-
-    private ScheduledFuture<?> lazyPollingScheduler;
-    private ScheduledFuture<?> fastPollingScheduler;
+    private @Nullable NeoHubConfiguration config;
+    private @Nullable NeoHubSocket socket;
+    private @Nullable ScheduledFuture<?> lazyPollingScheduler;
+    private @Nullable ScheduledFuture<?> fastPollingScheduler;
 
     private final AtomicInteger fastPollingCallsToGo = new AtomicInteger();
 
+    private Unit<?> temperatureUnit = SIUnits.CELSIUS;
+
+    private boolean isLegacyApiSelected = true;
+    private boolean isApiOnline = false;
+
+    private boolean systemDataDirty = true;
+    private long systemTimestamp = -1;
+    private Instant systemLastRefreshed = Instant.now().minusSeconds(3600);
+
+    private Map<String, Boolean> connectionStates = new HashMap<>();
```

Comment:
```suggestion
    private final Map<String, Boolean> connectionStates = new HashMap<>();
```

Also move this field above the non-final fields.

Reply:
Ok.